main.go

```diff
@@ -108,227 +101,208 @@ func runTests(machine *client.RobotClient) {
 	}
 
 	// wheeled base tests
-	logger.Info("Starting wheeled base tests...")
 	wb.baseTests(wheeledBase, odometry, wb.minLinVel, wb.minAngVel)
 
 	// sensor base tests
-	logger.Info("Starting sensor controlled base tests...")
 	sb.baseTests(sensorBase, odometry, sb.minLinVel, sb.minAngVel)
 
 	// encoded motor tests
-	logger.Info("Starting encoded motor tests...")
 	runMotorTests(leftMotor, odometry)
 
 	// controlled motor tests
-	logger.Info("Starting controlled motor tests...")
 	runMotorTests(rightMotor, odometry)
 
 	// single encoder tests
-	logger.Info("Starting encoder tests...")
 	runEncoderTests(leftMotor, leftEncoder)
+	runEncoderTests(rightMotor, rightEncoder)
 
 	// power sensor tests
-	logger.Info("Starting power sensor tests...")
 	runPowerSensorTests(powerSensor)
 
 	// movement sensor tests
-	logger.Info("Starting movement sensor tests...")
 	runMovementSensorTests(movementSensor)
 
 	// grid tests
-	logger.Info("Starting grid test with sensor controlled base...")
 	runGridTest(sensorBase, odometry)
 
-	if len(failedTests) == 0 {
-		logger.Info("All tests passed successfully!")
-	} else {
+	if len(failedTests) != 0 {
 		message := "tests failed: " + fmt.Sprint(len(failedTests)) + "/" + fmt.Sprint(totalTests) + "\n"
 		for i := 0; i < len(failedTests); i++ {
 			message += "- " + (failedTests[i] + "\n")
 		}
-		message += "for more info, see rovercanary.log"
 		sendSlackMessage(message)
```

Comment:
wondering if theres a way to send these to a thread rather than on the actual message. my worry is something goes wrong and 50 tests fail, so all 50 tests get posted into the channel. also worried about message limits but i doubt we would hit those

not a huge deal as its unlikely to happen, but would be nice to avoid spam/a wall of text

Reply:
I think it's pretty unlikely to happen, plus it will all me in the same message so it's not like going to send 55 messages it will just be a longer one